src/MongoDB.Driver.Core/Core/Bindings/CoreSession.cs

```diff
@@ -70,6 +87,100 @@ public sealed class CoreSession : ICoreSession
         public ICoreServerSession ServerSession => _serverSession;
 
         // public methods
+        /// <inheritdoc />
+        public void AbortTransaction(CancellationToken cancellationToken = default(CancellationToken))
+        {
+            EnsureIsInTransaction(nameof(AbortTransaction));
+
+            try
+            {
+                if (_currentTransaction.StatementId == 0)
+                {
+                    return;
+                }
+
+                try
+                {
+                    var firstAttempt = CreateAbortTransactionOperation();
+                    ExecuteEndTransactionOnPrimary(firstAttempt, cancellationToken);
+                    return;
+                }
+                catch (Exception exception) when (ShouldIgnoreAbortTransactionException(exception))
```

Comment:
I've written it this way rather than rely solely on the catch-all clause to ignore the exception because it's entirely possible that ShouldRetryEndTransactionException might ALSO return true for exceptions that should be ignored.